### PR TITLE
Add benchmark CI

### PR DESCRIPTION
**Pull request by ptoffy**
All credit goes to @gwynne

### CODE REVIEW

**Review by gwynne — APPROVED** [review]
Assuming it actually works as written (I couldn't exactly test it, after all!), approved 😛

File: .github/workflows/benchmark.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        run: sudo apt-get update && sudo apt-get install -y libjemalloc-dev
+      - name: Run benchmarks for PR branch
+        if: ${{ hashFiles('Benchmarks/**') != '' }}
+        run: |
+          swift package --allow-writing-to-directory .benchmarkBaselines/ benchmark baseline update pull_request
+          git checkout main
+          swift package --allow-writing-to-directory .benchmarkBaselines/ benchmark baseline update main
+          date >>"${GITHUB_STEP_SUMMARY}"
+          swift package benchmark baseline check main pull_request --format markdown >>"${GITHUB_STEP_SUMMARY}"
+        uses: thollander/actions-comment-pull-request@v2
```
Comment (gwynne) [inline .github/workflows/benchmark.yml:23]:
```suggestion
      - uses: thollander/actions-comment-pull-request@v2
```

**Review by gwynne — CHANGES_REQUESTED** [review]
For example, like that...

File: .github/workflows/benchmark.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    branches: [ main ]
+  
+jobs:
+  benchmark-delta-linux:
+    runs-on: ubuntu-latest
+    container: swift:jammy
+    steps:
+      - uses: actions/checkout@v4
+      - name: jemalloc dependency
+        run: sudo apt-get update && sudo apt-get install -y libjemalloc-dev
```
Comment (gwynne) [inline .github/workflows/benchmark.yml:14]:
```suggestion
        run: apt-get update && apt-get install -y libjemalloc-dev
```

**Review by 0xTim — APPROVED** [review]
LGTM - will let @gwynne approve as well

File: Benchmarks/Package.swift [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+import PackageDescription
+
+let package = Package(
+    name: "benchmarks",
+    platforms: [
+        .macOS(.v13),
+    ],
+    dependencies: [
+        .package(path: "../"),
```
Comment (gwynne) [inline Benchmarks/Package.swift:11]: I realize this is the way this tooling work, but it makes me shudder just the same 🤦‍♀️

**Review by gwynne — APPROVED** [review]

**Pull request merged**